### DISCUSSION TITLE
리팩토링 논의

### DISCUSSION THREAD

**Discussion by waterfogSW** [Student-Center/weave-server#107] [개발 논의]
- 리팩토링이 필요할것 같은 내용들을 태그로 언급해 주세요

**Comment by waterfogSW** [comment]
스크롤 조회 공통화관련 #106

**Reply by dojinyou** [reply]
스크롤 조회 공통화 할때 어노테이션 기반으로 하나의 클래스로 받을 수 있도록 정리되면 좋을 것 같아요

**Comment by dojinyou** [comment]
Query와 Command 분리하기

**Comment by dojinyou** [comment]
핵심 도메인의 식별자가 유일한 command일 때는 그냥 바로 인자로 사용하기

**Comment by waterfogSW** [comment]
#114 - 도메인간 이벤트 발행 분리

**Comment by waterfogSW** [comment]
userQueryUseCase, userGetByIdUseCase 중복

**Comment by waterfogSW** [comment]
UseCase postfix, GetUser

**Comment by waterfogSW** [comment]
application, domain service 통합?

**Comment by waterfogSW** [comment]
테스트 속도 올리기

**Comment by waterfogSW** [comment]
의존성 버전 관리 자동화 renovate 도입

**Comment by waterfogSW** [comment]
bootstrap 모듈에 event 패키지 만들기(도메인 이벤트 처리)

**Comment by waterfogSW** [comment]
회원 탈퇴시 OAuth provider측 데이터 제거 요청 추가 필요

**Comment by waterfogSW** [comment]
## getCurrentAuthentication 제거 리팩터링
- 유스케이스 파라미터로 받는게 아니라 securityContext에서 뒤적거려보는방식이라 명시적이지 않기도 하고, 매번 로직 구현이 강제됨
- controller 인자로 UserAuth 매핑하고, usecase 파라미터로 UserAuth넘기는 방식으로 개선
